shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -102,6 +108,34 @@ def _scan_location__check_stock(self, location):
             message = self.msg_store.location_empty(location)
             return self._response_for_select_location(message=message)
 
+    def _scan_location__check_package(self, location):
+        """Check if the location has lines without an assigned package."""
+        lines_without_package = self.env["stock.move.line"].search(
+            [
+                ("location_id", "=", location.id),
+                ("package_id", "=", False),
+                ("picking_id.picking_type_id", "in", self.picking_types.ids),
+            ]
+        )
+        if not lines_without_package:
+            message = (
+                self.msg_store.location_contains_only_lines_with_package_scan_one()
+            )
+            return self._response_for_select_location(message=message)
+
+    def _scan_package__check_location(self, package):
+        """Check if this package corresponds to any of the allowed locations."""
+        locations = self.picking_types.default_location_src_id
+        child_locations = self.env["stock.location"].search(
+            [("id", "child_of", locations.ids)]
+        )
+        allowed_locations = locations | child_locations
+        if package.location_id not in allowed_locations:
```

Comment:
Use instead `if not self.is_src_location_valid(package.location_id)`

Do the same change in `_scan_location__check_location`